src/test/java/org/breedinginsight/api/v1/controller/TraitUploadControllerIntegrationTest.java

```diff
@@ -78,11 +81,55 @@ public void setup() {
         // Insert user into program as not active
         dsl.execute(fp.get("InsertInactiveProgramUser"));
 
+        // Insert Trait
+        dsl.execute(fp.get("InsertProgramObservationLevel"));
+        dsl.execute(fp.get("InsertProgramOntology"));
+        dsl.execute(fp.get("InsertMethod"));
+        dsl.execute(fp.get("InsertScale"));
+        dsl.execute(fp.get("InsertTrait"));
+
         // Retrieve our new data
         validProgram = programDao.findAll().get(0);
 
     }
 
+    @Test
```

Comment:
Can we add a test for the case of an error in the TraitFileParser such as invalid scale and errors in other traits higher up in the service like missing required fields? Related to previous comment.

Reply:
Added test case